master/internal/resourcemanagers/agent_resource_manager.go

```diff
@@ -69,11 +69,18 @@ func (a *agentResourceManager) createResourcePool(
 	ctx *actor.Context, config *ResourcePoolConfig, cert *tls.Certificate,
 ) *actor.Ref {
 	ctx.Log().Infof("creating resource pool: %s", config.PoolName)
+
+	scheduler := a.config.Scheduler
```

Comment:
blocking: this logic might be better moved to `ResolveConfig` to avoid confusion.

Reply:
So if we put this logic into `ResolveConfig` when users print out their master configuration in the running cluster we would end up showing the `Scheduler` for every `resource_pool` not just the ones that are using a scheduler different than the one in `resource_manager`, which is also why I added an `omitempty` to the scheduler in `resource_pool`. 

I talked to @vishnu2kmohan this morning and confirmed that we want the following behavior when printing out the user config:
```
resource_manager:
  ...
  scheduler:
     type: "priority"
resource_pools:
  - pool_name: "hello"
```

versus having:
```
resource_manager:
  ...
  scheduler:
     "priority"
resource_pools:
  - pool_name: "hello"
    scheduler:
       type: "priority"
```